src/validation.h

```diff
@@ -409,7 +409,9 @@ enum DisconnectResult
     DISCONNECT_FAILED   // Something else went wrong.
 };
 
-class ConnectTrace;
+// forward declaration for ActivateBestChainStep and ConnectTip
+struct PerBlockConnectTrace;
+typedef std::vector<PerBlockConnectTrace> ConnectTrace;
```

Comment:
```suggestion
using ConnectTrace = std::vector<PerBlockConnectTrace>;
```

style nit for C++11

Reply:
I don't think it makes a difference for aliases which are not templates (and I can't see this in the style guide), but changed anyway.

Reply:
> I don't think it makes a difference for aliases which are not templates (and I can't see this in the style guide), but changed anyway.

This is my understanding as well after looking this up, along with `typedef` being the earlier, traditional syntax of the two. Perhaps one advantage `typedef` may have over `using` -- this is personal observation only and not for bikeshedding purposes :) -- is that it is used only for declaring type aliases.

Reply:
Git grepping instances of type aliases seems easier with`typedef` as well.